packages/db/src/lib/typeHelpers.ts

```diff
@@ -0,0 +1,19 @@
+import mapValues from 'lodash/mapValues';
+import isNull from 'lodash/isNull';
+
+const idFieldRegex = '^[a-zA-Z0-9_]*cumulus_id$';
+
+type RecordTypeWithNumberIdField<T> = {
+  [P in keyof T]: number | T[P];
+};
+
+type ArrayOfRecordTypeWithNumberIdField<T> = RecordTypeWithNumberIdField<T>[];
+
+export const convertIdFieldsToNumber = <T extends Record<string, any>>(record: T)
+  : RecordTypeWithNumberIdField<T> =>
+    mapValues(record, (value, key) =>
+      (key.match(idFieldRegex) && !isNull(value) ? Number(value) : value));
```

Comment:
Thinking about this, I wondering if we should be *explicit* about fields that match this rather than use a regular expression by convention.     

Or better yet, have the calling model/QuerySearchClient specify the field name instead of rely on regexp/committed code logic lists that need updating.  

I get why that will scatter this config around, but it also makes it more explicit which has less of a risk of accidental regexp matching.